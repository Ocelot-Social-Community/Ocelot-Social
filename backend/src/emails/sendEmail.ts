/* eslint-disable @typescript-eslint/no-unsafe-assignment */

/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-argument */
/* eslint-disable @typescript-eslint/restrict-template-expressions */
import path from 'node:path'

import Email from 'email-templates'
import { createTransport } from 'nodemailer'

// import type Email as EmailType from '@types/email-templates'

import CONFIG, { nodemailerTransportOptions } from '@config/index'
import logosWebapp from '@config/logosBranded'
import metadata from '@config/metadata'
import { UserDbProperties } from '@db/types/User'

const welcomeImageUrl = new URL(logosWebapp.LOGO_WELCOME_PATH, CONFIG.CLIENT_URI)
const settingsUrl = new URL('/settings/notifications', CONFIG.CLIENT_URI)

const defaultParams = {
  welcomeImageUrl,
  APPLICATION_NAME: CONFIG.APPLICATION_NAME,
  ORGANIZATION_NAME: metadata.ORGANIZATION_NAME,
  ORGANIZATION_URL: CONFIG.ORGANIZATION_URL,
  supportUrl: CONFIG.SUPPORT_URL,
  settingsUrl,
  renderSettingsUrl: true,
}

const from = `${CONFIG.APPLICATION_NAME} <${CONFIG.EMAIL_DEFAULT_SENDER}>`

const transport = createTransport(nodemailerTransportOptions)

const email = new Email({
  message: {
    from,
  },
  transport,
  i18n: {
    locales: ['en', 'de'],
    defaultLocale: 'en',
    retryInDefaultLocale: false,
    directory: path.join(__dirname, 'locales'),
    updateFiles: false,
    objectNotation: true,
    mustacheConfig: {
      tags: ['{', '}'],
      disable: false,
    },
  },
  send: CONFIG.SEND_MAIL,
  preview: false,
  // This is very useful to see the emails sent by the unit tests
  /*
  preview: {
    open: {
      app: 'brave-browser',
    },
  },
  */
})

interface OriginalMessage {
  to: string
  from: string
  attachments: string[]
  subject: string
  html: string
  text: string
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const sendNotificationMail = async (notification: any): Promise<OriginalMessage> => {
  const locale = notification?.to?.locale
  const to = notification?.email
  const name = notification?.to?.name
  const template = notification?.reason

  try {
    const { originalMessage } = await email.send({
      template: path.join(__dirname, 'templates', template),
      message: {
        to,
      },
      locals: {
        ...defaultParams,
        locale,
        name,
        postTitle:
          notification?.from?.__typename === 'Comment'
            ? notification?.from?.post?.title
            : notification?.from?.title,
        postUrl: new URL(
          notification?.from?.__typename === 'Comment'
            ? `/post/${notification?.from?.post?.id}/${notification?.from?.post?.slug}`
            : `/post/${notification?.from?.id}/${notification?.from?.slug}`,
          CONFIG.CLIENT_URI,
        ),
        postAuthorName:
          notification?.from?.__typename === 'Comment'
            ? undefined
            : notification?.from?.author?.name,
        postAuthorUrl:
          notification?.from?.__typename === 'Comment'
            ? undefined
            : new URL(
                `user/${notification?.from?.author?.id}/${notification?.from?.author?.slug}`,
                CONFIG.CLIENT_URI,
              ),
        commenterName:
          notification?.from?.__typename === 'Comment'
            ? notification?.from?.author?.name
            : undefined,
        commenterUrl:
          notification?.from?.__typename === 'Comment'
            ? new URL(
                `/user/${notification?.from?.author?.id}/${notification?.from?.author?.slug}`,
                CONFIG.CLIENT_URI,
              )
            : undefined,
        commentUrl:
          notification?.from?.__typename === 'Comment'
            ? new URL(
                `/post/${notification?.from?.post?.id}/${notification?.from?.post?.slug}#commentId-${notification?.from?.id}`,
                CONFIG.CLIENT_URI,
              )
            : undefined,
        // chattingUser: 'SR-71',
        // chatUrl: new URL('/chat', CONFIG.CLIENT_URI),
        groupUrl:
          notification?.from?.__typename === 'Group'
            ? new URL(
                `/group/${notification?.from?.id}/${notification?.from?.slug}`,
                CONFIG.CLIENT_URI,
              )
            : undefined,
        groupName:
          notification?.from?.__typename === 'Group' ? notification?.from?.name : undefined,
        groupRelatedUserName:
          notification?.from?.__typename === 'Group' ? notification?.relatedUser?.name : undefined,
        groupRelatedUserUrl:
          notification?.from?.__typename === 'Group'
            ? new URL(
                `/user/${notification?.relatedUser?.id}/${notification?.relatedUser?.slug}`,
                CONFIG.CLIENT_URI,
              )
            : undefined,
      },
    })
    return originalMessage as OriginalMessage
  } catch (error) {
    throw new Error(error)
  }
}

export interface ChatMessageEmailInput {
  senderUser: UserDbProperties
  recipientUser: UserDbProperties
  email: string
}

export const sendChatMessageMail = async (
  data: ChatMessageEmailInput,
): Promise<OriginalMessage> => {
  const { senderUser, recipientUser } = data
  const to = data.email
  try {
    const { originalMessage } = await email.send({
      template: path.join(__dirname, 'templates', 'chat_message'),
      message: {
        to,
      },
      locals: {
        ...defaultParams,
        locale: recipientUser.locale,
        name: recipientUser.name,
        chattingUser: senderUser.name,
        chattingUserUrl: new URL(`/user/${senderUser.id}/${senderUser.slug}`, CONFIG.CLIENT_URI),
        chatUrl: new URL('/chat', CONFIG.CLIENT_URI),
      },
    })
    return originalMessage as OriginalMessage
  } catch (error) {
    throw new Error(error)
  }
}

interface VerifyMailInput {
  email: string
  nonce: string
  locale: string
}

interface RegistrationMailInput extends VerifyMailInput {
  inviteCode?: string
}

export const sendRegistrationMail = async (
  data: RegistrationMailInput,
): Promise<OriginalMessage> => {
  const { nonce, locale, inviteCode } = data
  const to = data.email
  const actionUrl = new URL('/registration', CONFIG.CLIENT_URI)
  actionUrl.searchParams.set('email', to)
  actionUrl.searchParams.set('nonce', nonce)
  if (inviteCode) {
    actionUrl.searchParams.set('inviteCode', inviteCode)
    actionUrl.searchParams.set('method', 'invite-code')
  } else {
    actionUrl.searchParams.set('method', 'invite-mail')
  }

  try {
    const { originalMessage } = await email.send({
      template: path.join(__dirname, 'templates', 'registration'),
      message: {
        to,
      },
      locals: {
        ...defaultParams,
        locale,
        actionUrl,
        nonce,
        renderSettingsUrl: false,
      },
    })
    return originalMessage as OriginalMessage
  } catch (error) {
    throw new Error(error)
  }
}

interface EmailVerificationInput extends VerifyMailInput {
  name: string
}

export const sendEmailVerification = async (
  data: EmailVerificationInput,
): Promise<OriginalMessage> => {
  const { nonce, locale, name } = data
  const to = data.email
  const actionUrl = new URL('/settings/my-email-address/verify', CONFIG.CLIENT_URI)
  actionUrl.searchParams.set('email', to)
  actionUrl.searchParams.set('nonce', nonce)

  try {
    const { originalMessage } = await email.send({
      template: path.join(__dirname, 'templates', 'emailVerification'),
      message: {
        to,
      },
      locals: {
        ...defaultParams,
        locale,
        actionUrl,
        nonce,
        name,
        renderSettingsUrl: false,
      },
    })
    return originalMessage as OriginalMessage
  } catch (error) {
    throw new Error(error)
  }
}

export const sendResetPasswordMail = async (
  data: EmailVerificationInput,
): Promise<OriginalMessage> => {
  const { nonce, locale, name } = data
  const to = data.email
  const actionUrl = new URL('/password-reset/change-password', CONFIG.CLIENT_URI)
  actionUrl.searchParams.set('email', to)
  actionUrl.searchParams.set('nonce', nonce)
  try {
    const { originalMessage } = await email.send({
      template: path.join(__dirname, 'templates', 'resetPassword'),
      message: {
        to,
      },
      locals: {
        ...defaultParams,
        locale,
        actionUrl,
        nonce,
        name,
        renderSettingsUrl: false,
      },
    })
    return originalMessage as OriginalMessage
  } catch (error) {
    throw new Error(error)
  }
}

export const sendWrongEmail = async (data: {
  locale: string
  email: string
}): Promise<OriginalMessage> => {
  const { locale } = data
  const to = data.email
  const actionUrl = new URL('/password-reset/request', CONFIG.CLIENT_URI)
  try {
    const { originalMessage } = await email.send({
      template: path.join(__dirname, 'templates', 'wrongEmail'),
      message: {
        to,
      },
      locals: {
        ...defaultParams,
        locale,
        actionUrl,
        renderSettingsUrl: false,
      },
    })
    return originalMessage as OriginalMessage
  } catch (error) {
    throw new Error(error)
  }
}
