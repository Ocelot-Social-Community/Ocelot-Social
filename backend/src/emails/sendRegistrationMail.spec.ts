import { sendRegistrationMail } from './sendEmail'

describe('sendRegistrationMail', () => {
  const data: {
    email: string
    nonce: string
    locale: string
    inviteCode?: string
  } = {
    email: 'user@example.org',
    nonce: '123456',
    locale: 'en',
    inviteCode: 'welcome',
  }

  describe('with invite code', () => {
    describe('English', () => {
      beforeEach(() => {
        data.locale = 'en'
        data.inviteCode = 'welcome'
      })

      it('renders correctly', async () => {
        await expect(sendRegistrationMail(data)).resolves.toMatchSnapshot()
      })
    })

    describe('German', () => {
      beforeEach(() => {
        data.locale = 'de'
        data.inviteCode = 'welcome'
      })

      it('renders correctly', async () => {
        await expect(sendRegistrationMail(data)).resolves.toMatchSnapshot()
      })
    })
  })

  describe('without invite code', () => {
    describe('English', () => {
      beforeEach(() => {
        data.locale = 'en'
        delete data.inviteCode
      })

      it('renders correctly', async () => {
        await expect(sendRegistrationMail(data)).resolves.toMatchSnapshot()
      })
    })

    describe('German', () => {
      beforeEach(() => {
        data.locale = 'de'
        delete data.inviteCode
      })

      it('renders correctly', async () => {
        await expect(sendRegistrationMail(data)).resolves.toMatchSnapshot()
      })
    })
  })
})
