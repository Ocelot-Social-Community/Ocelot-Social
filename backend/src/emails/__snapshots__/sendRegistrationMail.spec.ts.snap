// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendRegistrationMail with invite code English renders correctly 1`] = `
{
  "attachments": [],
  "from": "ocelot.social <devops@ocelot.social>",
  "html": "<!DOCTYPE html>
<html lang="en">
  <head>
    <meta content="multipart/html; charset=UTF-8" http-equiv="content-type">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style>.wf-force-outline-none[tabindex="-1"]:focus{outline:none;}</style>
    <style>body{
  display: block;
  font-family: Lato, sans-serif;
  font-size: 17px;
  text-align: left;
  text-align: -webkit-left;
  justify-content: center;
  padding: 15px;
  margin: 0px;
}

h2 {
  margin-top: 25px;
  font-size: 25px;
  font-weight: normal;
  line-height: 22px;
  color: #333333;
}

.container {
  max-width: 680px;
  margin: 0 auto;
  display: block;
}

.head-logo {
  width: 60%;
  height: auto;
  display: block;
  margin-left: auto;
  margin-right: auto;
}

a {
  color: #17b53e;
}

a.button {
  background: #17b53e;
  font-family: Lato, sans-serif;
  font-size: 16px;
  line-height: 15px;
  text-decoration: none;
  text-align:center;
  padding: 13px 17px;
  color: #ffffff;
  display: table;
  margin-left: auto;
  margin-right: auto;
  border-radius: 4px;
}

span {
  color: #17b53e;
}

.text-block {
  margin-top: 20px;
  color: #000000;
}

footer {
  padding: 20px;
  font-family: Lato, sans-serif;
  font-size: 12px;
  line-height: 15px;
  text-align: center;
  color: #888888;
}

    </style>
  </head>
  <body>
    <div class="container">
      <header>
        <div class="head"><img class="head-logo" alt="Welcome Image" loading="lazy" src="http://webapp:3000/img/custom/logo-squared.svg">
        </div>
      </header>
      <h2>Welcome to ocelot.social!</h2>
      <div class="wrapper">
        <div class="content"></div>
        <p>Thank you for joining our cause – it's awesome to have you on board. There's just one tiny step missing before we can start shaping the world together … Please confirm your e-mail address by clicking the button below:</p><a class="button" href="http://webapp:3000/registration?email=user%40example.org&amp;nonce=123456&amp;inviteCode=welcome&amp;method=invite-code">Confirm your e-mail address</a>
        <p>If the above button doesn't work, you can also copy the following code into your browser window: <span>123456</span></p>
        <p>However, this only works if you have registered through our website.</p>
        <p>If you didn't sign up for <a>ocelot.social</a> we recommend you to check it out! It's a social network from people for people who want to connect and change the world together.
        </p>
        <p>PS: If you ignore this e-mail we will not create an account for you. ;)</p>
        <div class="text-block">
          <p>See you soon on <a class="organization" href="https://ocelot.social">ocelot.social</a>!</p>
          <p>– The ocelot.social Team</p>
        </div>
      </div>
      <footer>
        <div class="footer"></div><a href="https://ocelot.social">ocelot.social Community</a>
      </footer>
    </div>
  </body>
</html>",
  "subject": "Welcome to ocelot.social",
  "text": "WELCOME TO OCELOT.SOCIAL!

Thank you for joining our cause – it's awesome to have you on board. There's
just one tiny step missing before we can start shaping the world together …
Please confirm your e-mail address by clicking the button below:

Confirm your e-mail address
[http://webapp:3000/registration?email=user%40example.org&nonce=123456&inviteCode=welcome&method=invite-code]

If the above button doesn't work, you can also copy the following code into your
browser window: 123456

However, this only works if you have registered through our website.

If you didn't sign up for ocelot.social we recommend you to check it out! It's a
social network from people for people who want to connect and change the world
together.

PS: If you ignore this e-mail we will not create an account for you. ;)

See you soon on ocelot.social [https://ocelot.social]!

– The ocelot.social Team


ocelot.social Community [https://ocelot.social]",
  "to": "user@example.org",
}
`;

exports[`sendRegistrationMail with invite code German renders correctly 1`] = `
{
  "attachments": [],
  "from": "ocelot.social <devops@ocelot.social>",
  "html": "<!DOCTYPE html>
<html lang="de">
  <head>
    <meta content="multipart/html; charset=UTF-8" http-equiv="content-type">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style>.wf-force-outline-none[tabindex="-1"]:focus{outline:none;}</style>
    <style>body{
  display: block;
  font-family: Lato, sans-serif;
  font-size: 17px;
  text-align: left;
  text-align: -webkit-left;
  justify-content: center;
  padding: 15px;
  margin: 0px;
}

h2 {
  margin-top: 25px;
  font-size: 25px;
  font-weight: normal;
  line-height: 22px;
  color: #333333;
}

.container {
  max-width: 680px;
  margin: 0 auto;
  display: block;
}

.head-logo {
  width: 60%;
  height: auto;
  display: block;
  margin-left: auto;
  margin-right: auto;
}

a {
  color: #17b53e;
}

a.button {
  background: #17b53e;
  font-family: Lato, sans-serif;
  font-size: 16px;
  line-height: 15px;
  text-decoration: none;
  text-align:center;
  padding: 13px 17px;
  color: #ffffff;
  display: table;
  margin-left: auto;
  margin-right: auto;
  border-radius: 4px;
}

span {
  color: #17b53e;
}

.text-block {
  margin-top: 20px;
  color: #000000;
}

footer {
  padding: 20px;
  font-family: Lato, sans-serif;
  font-size: 12px;
  line-height: 15px;
  text-align: center;
  color: #888888;
}

    </style>
  </head>
  <body>
    <div class="container">
      <header>
        <div class="head"><img class="head-logo" alt="Welcome Image" loading="lazy" src="http://webapp:3000/img/custom/logo-squared.svg">
        </div>
      </header>
      <h2>Willkommen bei ocelot.social!</h2>
      <div class="wrapper">
        <div class="content"></div>
        <p>Danke, dass du dich angemeldet hast – wir freuen uns, dich dabei zu haben. Jetzt fehlt nur noch eine Kleinigkeit, bevor wir gemeinsam die Welt verbessern können … Bitte bestätige deine E-Mail Adresse:</p><a class="button" href="http://webapp:3000/registration?email=user%40example.org&amp;nonce=123456&amp;inviteCode=welcome&amp;method=invite-code">Bestätige deine E-Mail Adresse</a>
        <p>Sollte der Button für dich nicht funktionieren, kannst du auch folgenden Code in dein Browserfenster kopieren: <span>123456</span></p>
        <p>Das funktioniert allerdings nur, wenn du dich über unsere Website registriert hast.</p>
        <p>Falls du dich nicht selbst bei <a>ocelot.social</a> angemeldet hast, schau doch mal vorbei! Wir sind ein gemeinnütziges Aktionsnetzwerk – von Menschen für Menschen.
        </p>
        <p>PS: Wenn du keinen Account bei uns möchtest, kannst du diese E-Mail einfach ignorieren. ;)</p>
        <div class="text-block">
          <p>Bis bald bei <a class="organization" href="https://ocelot.social">ocelot.social</a>!</p>
          <p>– Dein ocelot.social Team</p>
        </div>
      </div>
      <footer>
        <div class="footer"></div><a href="https://ocelot.social">ocelot.social Community</a>
      </footer>
    </div>
  </body>
</html>",
  "subject": "Willkommen bei ocelot.social",
  "text": "WILLKOMMEN BEI OCELOT.SOCIAL!

Danke, dass du dich angemeldet hast – wir freuen uns, dich dabei zu haben. Jetzt
fehlt nur noch eine Kleinigkeit, bevor wir gemeinsam die Welt verbessern können
… Bitte bestätige deine E-Mail Adresse:

Bestätige deine E-Mail Adresse
[http://webapp:3000/registration?email=user%40example.org&nonce=123456&inviteCode=welcome&method=invite-code]

Sollte der Button für dich nicht funktionieren, kannst du auch folgenden Code in
dein Browserfenster kopieren: 123456

Das funktioniert allerdings nur, wenn du dich über unsere Website registriert
hast.

Falls du dich nicht selbst bei ocelot.social angemeldet hast, schau doch mal
vorbei! Wir sind ein gemeinnütziges Aktionsnetzwerk – von Menschen für Menschen.

PS: Wenn du keinen Account bei uns möchtest, kannst du diese E-Mail einfach
ignorieren. ;)

Bis bald bei ocelot.social [https://ocelot.social]!

– Dein ocelot.social Team


ocelot.social Community [https://ocelot.social]",
  "to": "user@example.org",
}
`;

exports[`sendRegistrationMail without invite code English renders correctly 1`] = `
{
  "attachments": [],
  "from": "ocelot.social <devops@ocelot.social>",
  "html": "<!DOCTYPE html>
<html lang="en">
  <head>
    <meta content="multipart/html; charset=UTF-8" http-equiv="content-type">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style>.wf-force-outline-none[tabindex="-1"]:focus{outline:none;}</style>
    <style>body{
  display: block;
  font-family: Lato, sans-serif;
  font-size: 17px;
  text-align: left;
  text-align: -webkit-left;
  justify-content: center;
  padding: 15px;
  margin: 0px;
}

h2 {
  margin-top: 25px;
  font-size: 25px;
  font-weight: normal;
  line-height: 22px;
  color: #333333;
}

.container {
  max-width: 680px;
  margin: 0 auto;
  display: block;
}

.head-logo {
  width: 60%;
  height: auto;
  display: block;
  margin-left: auto;
  margin-right: auto;
}

a {
  color: #17b53e;
}

a.button {
  background: #17b53e;
  font-family: Lato, sans-serif;
  font-size: 16px;
  line-height: 15px;
  text-decoration: none;
  text-align:center;
  padding: 13px 17px;
  color: #ffffff;
  display: table;
  margin-left: auto;
  margin-right: auto;
  border-radius: 4px;
}

span {
  color: #17b53e;
}

.text-block {
  margin-top: 20px;
  color: #000000;
}

footer {
  padding: 20px;
  font-family: Lato, sans-serif;
  font-size: 12px;
  line-height: 15px;
  text-align: center;
  color: #888888;
}

    </style>
  </head>
  <body>
    <div class="container">
      <header>
        <div class="head"><img class="head-logo" alt="Welcome Image" loading="lazy" src="http://webapp:3000/img/custom/logo-squared.svg">
        </div>
      </header>
      <h2>Welcome to ocelot.social!</h2>
      <div class="wrapper">
        <div class="content"></div>
        <p>Thank you for joining our cause – it's awesome to have you on board. There's just one tiny step missing before we can start shaping the world together … Please confirm your e-mail address by clicking the button below:</p><a class="button" href="http://webapp:3000/registration?email=user%40example.org&amp;nonce=123456&amp;method=invite-mail">Confirm your e-mail address</a>
        <p>If the above button doesn't work, you can also copy the following code into your browser window: <span>123456</span></p>
        <p>However, this only works if you have registered through our website.</p>
        <p>If you didn't sign up for <a>ocelot.social</a> we recommend you to check it out! It's a social network from people for people who want to connect and change the world together.
        </p>
        <p>PS: If you ignore this e-mail we will not create an account for you. ;)</p>
        <div class="text-block">
          <p>See you soon on <a class="organization" href="https://ocelot.social">ocelot.social</a>!</p>
          <p>– The ocelot.social Team</p>
        </div>
      </div>
      <footer>
        <div class="footer"></div><a href="https://ocelot.social">ocelot.social Community</a>
      </footer>
    </div>
  </body>
</html>",
  "subject": "Welcome to ocelot.social",
  "text": "WELCOME TO OCELOT.SOCIAL!

Thank you for joining our cause – it's awesome to have you on board. There's
just one tiny step missing before we can start shaping the world together …
Please confirm your e-mail address by clicking the button below:

Confirm your e-mail address
[http://webapp:3000/registration?email=user%40example.org&nonce=123456&method=invite-mail]

If the above button doesn't work, you can also copy the following code into your
browser window: 123456

However, this only works if you have registered through our website.

If you didn't sign up for ocelot.social we recommend you to check it out! It's a
social network from people for people who want to connect and change the world
together.

PS: If you ignore this e-mail we will not create an account for you. ;)

See you soon on ocelot.social [https://ocelot.social]!

– The ocelot.social Team


ocelot.social Community [https://ocelot.social]",
  "to": "user@example.org",
}
`;

exports[`sendRegistrationMail without invite code German renders correctly 1`] = `
{
  "attachments": [],
  "from": "ocelot.social <devops@ocelot.social>",
  "html": "<!DOCTYPE html>
<html lang="de">
  <head>
    <meta content="multipart/html; charset=UTF-8" http-equiv="content-type">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style>.wf-force-outline-none[tabindex="-1"]:focus{outline:none;}</style>
    <style>body{
  display: block;
  font-family: Lato, sans-serif;
  font-size: 17px;
  text-align: left;
  text-align: -webkit-left;
  justify-content: center;
  padding: 15px;
  margin: 0px;
}

h2 {
  margin-top: 25px;
  font-size: 25px;
  font-weight: normal;
  line-height: 22px;
  color: #333333;
}

.container {
  max-width: 680px;
  margin: 0 auto;
  display: block;
}

.head-logo {
  width: 60%;
  height: auto;
  display: block;
  margin-left: auto;
  margin-right: auto;
}

a {
  color: #17b53e;
}

a.button {
  background: #17b53e;
  font-family: Lato, sans-serif;
  font-size: 16px;
  line-height: 15px;
  text-decoration: none;
  text-align:center;
  padding: 13px 17px;
  color: #ffffff;
  display: table;
  margin-left: auto;
  margin-right: auto;
  border-radius: 4px;
}

span {
  color: #17b53e;
}

.text-block {
  margin-top: 20px;
  color: #000000;
}

footer {
  padding: 20px;
  font-family: Lato, sans-serif;
  font-size: 12px;
  line-height: 15px;
  text-align: center;
  color: #888888;
}

    </style>
  </head>
  <body>
    <div class="container">
      <header>
        <div class="head"><img class="head-logo" alt="Welcome Image" loading="lazy" src="http://webapp:3000/img/custom/logo-squared.svg">
        </div>
      </header>
      <h2>Willkommen bei ocelot.social!</h2>
      <div class="wrapper">
        <div class="content"></div>
        <p>Danke, dass du dich angemeldet hast – wir freuen uns, dich dabei zu haben. Jetzt fehlt nur noch eine Kleinigkeit, bevor wir gemeinsam die Welt verbessern können … Bitte bestätige deine E-Mail Adresse:</p><a class="button" href="http://webapp:3000/registration?email=user%40example.org&amp;nonce=123456&amp;method=invite-mail">Bestätige deine E-Mail Adresse</a>
        <p>Sollte der Button für dich nicht funktionieren, kannst du auch folgenden Code in dein Browserfenster kopieren: <span>123456</span></p>
        <p>Das funktioniert allerdings nur, wenn du dich über unsere Website registriert hast.</p>
        <p>Falls du dich nicht selbst bei <a>ocelot.social</a> angemeldet hast, schau doch mal vorbei! Wir sind ein gemeinnütziges Aktionsnetzwerk – von Menschen für Menschen.
        </p>
        <p>PS: Wenn du keinen Account bei uns möchtest, kannst du diese E-Mail einfach ignorieren. ;)</p>
        <div class="text-block">
          <p>Bis bald bei <a class="organization" href="https://ocelot.social">ocelot.social</a>!</p>
          <p>– Dein ocelot.social Team</p>
        </div>
      </div>
      <footer>
        <div class="footer"></div><a href="https://ocelot.social">ocelot.social Community</a>
      </footer>
    </div>
  </body>
</html>",
  "subject": "Willkommen bei ocelot.social",
  "text": "WILLKOMMEN BEI OCELOT.SOCIAL!

Danke, dass du dich angemeldet hast – wir freuen uns, dich dabei zu haben. Jetzt
fehlt nur noch eine Kleinigkeit, bevor wir gemeinsam die Welt verbessern können
… Bitte bestätige deine E-Mail Adresse:

Bestätige deine E-Mail Adresse
[http://webapp:3000/registration?email=user%40example.org&nonce=123456&method=invite-mail]

Sollte der Button für dich nicht funktionieren, kannst du auch folgenden Code in
dein Browserfenster kopieren: 123456

Das funktioniert allerdings nur, wenn du dich über unsere Website registriert
hast.

Falls du dich nicht selbst bei ocelot.social angemeldet hast, schau doch mal
vorbei! Wir sind ein gemeinnütziges Aktionsnetzwerk – von Menschen für Menschen.

PS: Wenn du keinen Account bei uns möchtest, kannst du diese E-Mail einfach
ignorieren. ;)

Bis bald bei ocelot.social [https://ocelot.social]!

– Dein ocelot.social Team


ocelot.social Community [https://ocelot.social]",
  "to": "user@example.org",
}
`;
